yt/fields/species_fields.py

```diff
@@ -208,38 +208,50 @@ def add_nuclei_density_fields(registry, ftype):
 
     # Here, we add default nuclei and number density fields for H and
     # He if they are not defined above. This assumes full ionization!
-    for element in ["H", "He"]:
-        if element in elements:
-            continue
-        registry.add_field(
-            (ftype, f"{element}_nuclei_density"),
-            sampling_type="local",
-            function=_default_nuclei_density,
-            units=unit_system["number_density"],
-        )
-        if element == "H":
-            registry.alias((ftype, "H_p1_number_density"), (ftype, "H_nuclei_density"))
-
-        if element == "He":
-            registry.alias(
-                (ftype, "He_p2_number_density"), (ftype, "He_nuclei_density")
+    if registry.ds.default_species_fields is not None:
+        dsf = registry.ds.default_species_fields
+        for element in ["H", "He"]:
+            if element in elements:
+                continue
+            registry.add_field(
+                (ftype, f"{element}_nuclei_density"),
+                sampling_type="local",
+                function=_default_nuclei_density,
+                units=unit_system["number_density"],
+            )
+            if element == "H":
```

Comment:
Can you leave comments here explaining?  I mean, *obviously* I totally understand and don't need you to explain this to me, but, you know, in case someone else came across it, like my friend, his username is tatthewmurk I think, if he wanted to have a better understanding, yeah?

Reply:
Tell tatthewmurk that we've got some comments in there now :)